.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,62 @@
+<--
+### Contribution Checklist
+  
+  - Name the pull request in the form "[component] Title of the pull request". Skip *component* if you are unsure about which is the best component. E.g. `[docs] Fix typo in produce method`.
+
+  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
+  
+  - Make sure that the change passes the CI checks.
+
+  - Each pull request should address only one issue, not mix up code from multiple issues.
+  
+  - Each commit in the pull request has a meaningful commit message
+
+  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.
```

Comment:
Why don't we use actual checkboxes?
- [ ] abc
- [ ] xyz

Reply:
good point. I will update

Reply:
No I mean all the CI checks of the PR are green. How to build and test the PR is covered by the contribution guide in the website. The template only acts a checklist for reminding contributors to do things to make sure PR is in a good shape for reviewers to review.

Reply:
> No I mean all the CI checks of the PR are green.

A PR is not created and CI checks are not run yet when a contributor read it. To remove the (invisible) checklist AFTER all items of the checklist are addressed, the contributor need to edit the PR's description. I don't think contributors do it. This is a reason I suggested using actual check box.

Maybe we can say "all PRs need to pass all CI checks" as a note instead of having it as an item of the check list if it's just a reminder.